OBSDOCS-3318 [DOC] Documentation for Perses Dashboard (#111992)
openshift/openshift-docs: modules/coo-supported-perses-plugins.adoc

// Module included in the following assemblies:

// * observability/cluster_observability_operator/ui_plugins/perses-dashboard.adoc

:_mod-docs-content-type: REFERENCE
[id="coo-supported-perses-plugins_{context}"]
= Supported Perses plugins

[role="_abstract"]

> The following tables list the Perses plugins available in {product-title} and indicate whether each plugin is supported in the current release.

Reviewer: I think it will be easier to point out which plugins are actually supported, as the list is constantly growing.
```suggestion
The following tables list the supported Perses plugins in {product-title}.
```

 
Plugins are organized by type: panel, datasource, query, variable, and explore.
 
== Panel plugins
 
Panel plugins define the visualization types available when building dashboards.
 

> [cols="1,1",options="header"]
> |===
> |Plugin | Supported
>  
> |Bar Chart
> |Yes
>  
> |Flame Chart
> |Yes
>  
> |Gauge Chart
> |Yes
>  
> |Heat Map Chart
> |Yes
>  
> |Histogram Chart
> |Yes
>  
> |Logs Table
> |Yes
>  
> |Markdown
> |Yes
>  
> |Pie Chart
> |Yes
>  
> |Scatter Chart
> |Yes
>  
> |Stat Chart
> |Yes
>  
> |Status History Chart
> |Yes
>  
> |Table
> |Yes
>  
> |Time Series Chart
> |Yes
>  
> |Time Series Table
> |Yes
>  
> |Trace Table
> |Yes
>  
> |Tracing Gantt Chart
> |Yes
> |===

Reviewer: ```suggestion
[cols="1",options="header"]
|===
|Plugin
 
|Bar Chart
 
|Flame Chart
 
|Gauge Chart
 
|Heat Map Chart
 
|Histogram Chart
 
|Logs Table
 
|Markdown
 
|Pie Chart
 
|Scatter Chart
 
|Stat Chart
 
|Status History Chart
 
|Table
 
|Time Series Chart
 
|Time Series Table
 
|Trace Table
 
|Tracing Gantt Chart
|===
```

Author: I am not removing the "Supported" column since it will be needed for doc consistency.

 
== Datasource plugins
 
Datasource plugins define the backends that Perses can query.
 
[cols="1,1",options="header"]
|===
|Plugin | Supported
 
|ClickHouse Datasource
|No
 

> |Loki Datasource
> |Yes
>  
> |Prometheus Datasource
> |Yes
>  
> |Pyroscope Datasource
> |No
>  
> |Tempo Datasource
> |Yes
>  
> |Victoria Logs Datasource
> |No
> |===

Reviewer: ```suggestion
Datasource plugins define the backends that Perses can query.
 
[cols="1",options="header"]
|===
|Plugin
 
|Loki Datasource
 
|Prometheus Datasource
 
|Tempo Datasource
|===
```

 
== Query plugins
 
Query plugins define the query types that panels can use to retrieve data from a datasource.
 
[cols="1,1,1",options="header"]
|===
|Plugin | Query type | Supported
 
|ClickHouse Log Query
|Log Query
|No
 
|ClickHouse Time Series Query
|Time Series Query
|No
 
|Loki Log Query
|Log Query
|Yes
 
|Loki Time Series Query
|Time Series Query
|Yes
 
|Prometheus Time Series Query
|Time Series Query
|Yes
 

> |Pyroscope Profile Query
> |Profile Query
> |No
>  
> |Tempo Trace Query
> |Trace Query
> |Yes
>  
> |Victoria Logs Query
> |Log Query
> |No
>  
> |Victoria Logs Time Series Query
> |Time Series Query
> |No
> |===

Reviewer: ```suggestion
[cols="1,1",options="header"]
|===
|Plugin | Query types
 
|Loki
|Log Query, Time Series Query
 
|Prometheus
|Time Series Query
 
|Tempo Trace Query
|Trace Query
 
|===
```

 
== Variable plugins
 
Variable plugins define the dynamic variable types available in dashboards. Variables allow users to filter and scope dashboard data interactively.
 
[cols="1,1",options="header"]
|===

> |Plugin | Supported
>  
> |Datasource Variable
> |Yes
>  
> |Loki Label Names Variable
> |Yes
>  
> |Loki Label Values Variable
> |Yes
>  
> |Loki LogQL Variable
> |Yes
>  
> |Prometheus Label Names Variable
> |Yes
>  
> |Prometheus Label Values Variable
> |Yes
>  
> |Prometheus PromQL Variable
> |Yes
>  
> |Static List Variable
> |Yes
>  
> |Victoria Logs Field Names Variable
> |No
>  
> |Victoria Logs Field Values Variable
> |No
> |===

Reviewer: ```suggestion
== Variable plugins
 
Variable plugins define the dynamic variable types available in dashboards. Variables allow users to filter and scope dashboard data interactively.
 
[cols="1",options="header"]
|===
|Plugin
 
|Datasource Variable
 
|Loki Label Names Variable
 
|Loki Label Values Variable
 
|Loki LogQL Variable
 
|Prometheus Label Names Variable
 
|Prometheus Label Values Variable
 
|Prometheus PromQL Variable
 
|Static List Variable
 
|===
```

 
== Explore plugins
 
Explore plugins provide interactive query exploration interfaces within Perses.
 
[cols="1,1",options="header"]
|===
|Plugin | Supported
 
|Prometheus Explorer
|No
 
|Pyroscope Explorer
|No
 
|Tempo Traces Explorer
|No
|===
 
[NOTE]
====
Plugins listed as not supported are available in upstream Perses but are not
supported in this release of {product-title}. Support status may change in future releases.

> ====

Reviewer: ```suggestion
Other plugins not listed above might be available but are not
supported in this release of {product-title}. Support status may change in future releases.
====
```
